Client/Frontend/Browser/SavedTab+ConfigureExtension.swift

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import WebKit
+import Storage
+import Shared
+
+// This cannot be easily imported into extension targets, so we break it out here.
+extension SavedTab {
+    convenience init?(tab: Tab, isSelected: Bool) {
+        assert(Thread.isMainThread)
+        
+        var sessionData = tab.sessionData
+        if sessionData == nil {
+            let currentItem: WKBackForwardListItem! = tab.webView?.backForwardList.currentItem
+
+            // Freshly created web views won't have any history entries at all.
+            // If we have no history, abort.
+            if currentItem != nil {
+                let backList = tab.webView?.backForwardList.backList ?? []
```

Comment:
A single line comment on what back, forward list is trying to do be be nice for any future reader

Reply:
Done :)